src/agent.rs

```diff
@@ -171,35 +142,16 @@ impl Agent {
     pub fn put(&self, path: &str) -> Request {
         self.request("PUT", path)
     }
-
-    /// Make a DELETE request from this agent.
-    pub fn delete(&self, path: &str) -> Request {
-        self.request("DELETE", path)
-    }
-
```

Comment:
I think we should keep DELETE. since without it one can't use a classic REST API (CRUD)